_posts/2020-04-27-introducing-command-mode.adoc

```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: 'Introducing Command Mode'
+date: 2020-04-27
+tags: development-tips 
+author: maxandersen
+---
+
+Ever wanted to use Quarkus awesome API's and full feature set from a command line application ? 
+
+Did you need to run a scheduled batch job now and then and not wanting to embed it into your main Quarkus built service ? 
+
+image:https://i.imgflip.com/3xwzsh.jpg[]
+
+Quarkus has thus far been used to write applications that starts an endpoint (i.e. webserver via http) and then runs for a long time.
+
+In Quarkus 1.4 command mode let you write apps that runs without an endpoint and optionally exits immediatly.
+
+This enables you to use Quarkus for writing a whole new style of appliations - think command line clients (CLI), batch scripts, console apps, etc.
+
+== How to use it
+
+Below is a simple `GreetingMain` class which uses the traditional `GreetingService` from all our quickstarts.
+
+[source,java]
+----
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.annotations.QuarkusMain;
+
+@QuarkusMain    // <.>
+public class GreetingMain implements QuarkusApplication {
+  @Inject // <.>
+  GreetingService service;
+
+  @Override
+  public int run(String... args) throws Exception {   // <.>
+
+    if(arg.length>0) {
+      System.out.println(service(arg[0]);
+    } else P{
+      System.out.println(service(""));
+    }
+
+    return 0;
+ }
+}
+----
+<.> The `@QuarkusMain` annotation tells Quarkus that this is the main entry point.
+<.> The `run` method is invoked once Quarkus starts, and the application stops when it finishes. If you would like to access request scoped beans you could annotate it with `@ActivateRequestContext`.
+
+You can add this to a `GreetingMain.java` and compile to a .jar or to full native (using `mvnw package -Dnative`) and when run you get something like:
+
+```
+ ./target/getting-started-command-mode-1.0-SNAPSHOT-runner commando
+__  ____  __  _____   ___  __ ____  ______
+ --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) getting-started-command-mode 1.0-SNAPSHOT (powered by Quarkus 999-SNAPSHOT) started in 0.005s.
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) Profile prod activated.
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) Installed features: [cdi]
+hello commando
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) getting-started-command-mode stopped in 0.000s
+```
+
+=== Development mode
+
+Additionally command mode also works with `quarkus:dev` aka. dev-mode.
+
+When you run with `mvn quarkus:dev` you can add `-Dquarkus.args=yourvalue` as arguments you want to pass into the command line. 
```

Comment:
explain how to do multiple arguments (I image with "some bar") and how to escape `"` :)

Reply:
illustrated in example following this.